ext/mbstring/mbstring.c

```diff
@@ -2537,7 +2537,7 @@ static zend_string* mb_trim_string(zend_string *input, zend_string *marker, cons
 		if (out_len <= to_skip) {
 			to_skip -= out_len;
 		} else {
-			for (int i = to_skip; i < out_len; i++) {
+			for (unsigned int i = to_skip; i < out_len; i++) {
```

Comment:
Strictly speaking, there is an inconsistency here with how you did the previous changes. Previously you changed this to size_t because out_len was size_t. Here you changed it to unsigned int.
unsigned int, or even int, suffices in all cases because the length is limited to 128 anyway. Performance-wise it's hard to predict what these would do. I would expect no difference because it would just use a larger register , albeit with a larger stack spill.

Reply:
> Strictly speaking, there is an inconsistency here with how you did the previous changes. Previously you changed this to size_t because out_len was size_t. Here you changed it to unsigned int. unsigned int, or even int, suffices in all cases because the length is limited to 128 anyway. Performance-wise it's hard to predict what these would do. I would expect no difference because it would just use a larger register , albeit with a larger stack spill.

Right, I did _kinda_ blindly do those changes. But if those are limited to 128, might as well use an unsigned char then?

Reply:
Okay I remember how I got to this inconsistency, I used the type of the ``from`` parameter which is ``unsigned int`` here. But I wonder if this might not cause issues and that should be of type ``size_t`` in the first place?

Reply:
The from variable should probably be of type size_t, otherwise >4GiB strings (okay unlikely, but still) won't work correctly I think.